app/jobs/clean_abandoned_site_channels_job.rb

```diff
@@ -6,15 +6,13 @@ def perform
     # and the publisher's session has expired. This ensures the channel will not be visible to them again and can
     # be safely deleted.
 
-    channel_details = SiteChannelDetails.abandoned
+    channels = Channel.not_visible_site_channels
     n = 0
-    channel_details.each do |details|
-      raise unless details.verification_token.nil?
-
-      details.channel.destroy
+    channels.joins(:site_channel_details).each do |channel|
+      raise unless channel.details.verification_method.nil?
```

Comment:
can we change this to 
`raise if channel.details.verification_method.present?` 
i think that's slightly more readable